src/blr/cli/tagfastq.py

```diff
@@ -46,31 +46,32 @@ def main(args):
     out_interleaved = not args.output2
     logger.info(f"Output detected as {'interleaved' if out_interleaved else 'paired'} FASTQ.")
 
-    raw_barcodes_cache = dict()
+    uncorrected_barcodes_cache = dict()
     reads_missing_barcode = 0
     separator = args.sep
-    # Parse input FASTA/FASTQ for read1 and read2, raw barcodes and write output
+    # Parse input FASTA/FASTQ for read1 and read2, uncorrected barcodes and write output
     with dnaio.open(args.input1, file2=args.input2, interleaved=in_interleaved, mode="r",
                     fileformat="fastq") as reader, \
             dnaio.open(args.output1, file2=args.output2, interleaved=out_interleaved, mode="w",
                        fileformat="fastq") as writer, \
-            dnaio.open(args.raw_barcodes, mode="r") as raw_bc_reader:
+            dnaio.open(args.uncorrected_barcodes, mode="r") as uncorrected_barcode_reader:
 
-        raw_bc_iterator = parse_raw_barcodes(raw_bc_reader)
+        uncorrected_barcode_iterator = parse_uncorrected_barcodes(uncorrected_barcode_reader)
 
         for read1, read2 in tqdm(reader, desc="Read pairs processed"):
             # Header parsing
             name_and_pos_r1, read_and_index_r1 = read1.name.split(maxsplit=1)
             name_and_pos_r2, read_and_index_r2 = read2.name.split(maxsplit=1)
 
-            raw_barcode_seq, raw_barcodes_cache = search_bc(raw_bc_iterator, name_and_pos_r1, raw_barcodes_cache)
+            uncorrected_barcode_seq = search_barcode(uncorrected_barcode_iterator, name_and_pos_r1,
```

Comment:
Have you considered factoring out a class? This is totally optional and I just want to mention it since I noticed it now, but if you create a class `BarcodeReader`/`BarcodeCache` or similar, then you could move the cache dict, the iterator, the barcode parser and the `search_barcode` function into it. This would nicely encapsulate the barcode reading functionality. From the main function, you would only need to call the constructor and then call `barcode_cache.retrieve(name_and_pos_r1)` or similar.

Reply:
I did an attempt on this. I have not work much with classes before, so if I am doing something bad or against conventions please inform me :)

Reply:
Looks quite good already, I’ll just add a couple of comments